Public/Src/Engine/Processes/SandboxedProcessPipExecutor.cs

```diff
@@ -3736,5 +3773,28 @@ private void PreparePathForDirectory(string expandedDirectoryPath, bool createIf
                 FileUtilities.CreateDirectory(expandedDirectoryPath);
             }
         }
+
+        private bool ShouldPreserveOutput(AbsolutePath path, HashSet<AbsolutePath> whitelist)

```

Comment:
>ShouldPreserveOutput [](start = 21, length = 20)

nit - this is only valid on declared output files and declared output directories. It would not be valid if called on a file contained within an output directory, unlike the method in PipArtifacts. I suggest adding some comments or modifying the function name to make this clear.

Reply:
agree, it is confusing. 

renamed to ShouldPreserveDeclaredOutput and added comments. 

---
In reply to: [287430865](https://github.com/microsoft/BuildXL/pull/369#discussion_r287430865) [](ancestors = 287430865)